pac4j-saml-db/src/main/java/org/pac4j/saml/dbclient/dao/api/SamlClientDao.java

```diff
@@ -0,0 +1,22 @@
+package org.pac4j.saml.dbclient.dao.api;
+
+import java.util.List;
+
+import org.pac4j.saml.dbclient.DbLoadedSamlClientConfiguration;
+
+
+/**
+ * DAO to manipulate SAML Client Configurations stored in a data source.
+ * 
+ * @author jkacer
+ */
+public interface SamlClientDao {
+
+	/**
+	 * Loads all existing SAML Client Configurations.
+	 * 
+	 * @return A list of configurations.
+	 */
+	public List<DbLoadedSamlClientConfiguration> loadAllClients();

```

Comment:
While it's better to load all clients at the same time, I'd prefer to load each client's configuration separately to embed this logic inside the SAML database configuration component.
You could even keep this "load all" logic, cache the result and just provide configuration one by one for each `SAML2Client`.